substrate-node/pallets/pallet-tfgrid/src/lib.rs

```diff
@@ -1822,11 +1823,21 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             T::RestrictedOrigin::ensure_origin(origin)?;
 
-            let mut farm = Farms::<T>::get(farm_id).ok_or(Error::<T>::FarmNotExists)?;
-            farm.farming_policy_limits = limits.clone();
-            Farms::<T>::insert(farm_id, farm);
+            if let Some(policy_limits) = limits {
+                let farming_policy = FarmingPoliciesMap::<T>::get(policy_limits.farming_policy_id);
+                let now = system::Pallet::<T>::block_number();
+
+                if now >= farming_policy.policy_created + farming_policy.policy_end {
```

Comment:
Why created + end? Logic would dictate that only end is sufficient?

Reply:
`policy_end` is specified in number of blocks and not some block in the future

Reply:
In that case, please add a comment to indicate this as it won't be clear for readers who come accros this code